cli/azd/pkg/environment/environment.go

```diff
@@ -57,19 +59,36 @@ type Environment struct {
 	Config config.Config
 }
 
+const AzureInitialEnvironmentConfigName = "AZURE_INITIAL_ENVIRONMENT_CONFIG"
+
 // New returns a new environment with the specified name.
 func New(name string) *Environment {
 	env := &Environment{
 		name:        name,
 		dotenv:      make(map[string]string),
 		deletedKeys: make(map[string]struct{}),
-		Config:      config.NewEmptyConfig(),
+		Config:      getInitialConfig(),
 	}
 
 	env.SetEnvName(name)
 	return env
 }
 
+func getInitialConfig() config.Config {
+	initialConfig := os.Getenv(AzureInitialEnvironmentConfigName)
+	if initialConfig == "" {
+		return config.NewEmptyConfig()
+	}
+
+	var initConfig map[string]any
+	if err := json.Unmarshal([]byte(initialConfig), &initConfig); err != nil {
+		log.Panicln("Failed to unmarshal initial config", err, "Using empty config.")
```

Comment:
This will actually end up calling `panic`, which mean's we won't actually use the empty config.  I feel like `panic`ing here is wrong.  Maybe this should be a log.Print?